roles/cifmw_cephadm/tasks/post.yml

```diff
@@ -70,7 +78,9 @@
       oc rsh openstackclient openstack endpoint create --region regionOne object-store internal http://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
       oc rsh openstackclient openstack role add --project admin --user admin swiftoperator
   delegate_to: localhost
-  when: swift_endpoints_count.stdout == "0"
+  when:
+    - not swift_in_ctlplane.stdout | bool
+    - swift_endpoints_count.stdout == "0"
```

Comment:
I think you're on the right track. Maybe we want something like this:
```
fail:
  msg: "WARNING: Swift is already deployed so you cannot deploy RGW"
when:
- cifmw_ceph_daemons_layout.rgw_enabled | bool
- swift_in_ctlplane.stdout | bool
- swift_endpoints_count.stdout > 0
```
or maybe change from `failure` to `debug`

Reply:
and move that warning earlier in the ceph.yml playbook so we fail early before doing the deployment.

Reply:
In the latest patch i moved the warning to pre.yaml but i updated the msg as
 "WARNING: Swift is deployed and the endpoint exists already, ceph RGW cannot be configured as object store"

because we always deploy rgw here [1]
iiuc,  in this PR we are just configuring the deployed rgw as the object store at the end.

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L268

Reply:
hi @katarimanojk - Just wondering if we should add this openstack configuration inside cifmw_cephadm role. Is there anything that don't allow us to do this configuration outside of the role?

Reply:
@viroel has a valid point. `cifmw_cephadm` is already a big role and it's about configuring ceph not openstack, but these changes configure openstack.

You could introduce a separate role like `cifmw_obj_store_keystone` (since this ultimately results in changes to keystone so that the object store is registered). You would then call this role from the ceph.yml playbook after `Render Ceph client configuration`. When the new role is called, you would have to pass variables which were defined by `cifmw_cephadm` like `{{ cifmw_ceph_rgw_keystone_psw }}` and `{{ cifmw_cephadm_vip }}`.

https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L316

Because we need `cifmw_cephadm` to not depend on k8s, it's important that we only call these tasks if a k8s env (`cifmw_openshift_kubeconfig`) is present. I would require that condition. I.e. the `cifmw_obj_store_keystone`

The role itself can be simple. main.yml can basically include the items being added to pre.yml and post.yml.

Reply:
I'm not sure it's good to have yet another role for a couple of tasks but we can definitely talk more about that if it's the desired direction.
In general I think that if `Swift` is not detected in the ctlplane (e.g. VA1, but in `architectures` is disabled by default), we might need to be able to configure a Ceph service (`RGW`) in keystone, but I think we should be responsible to do such configuration here, because we have all the variables in a single place, and we don't risk to complicate more the flow.
I would propose a follow up if we think about a refactoring of this patch, but personally I like to have a `post.yaml` when we perform some post-ceph operations.

Reply:
OK, we don't need to block on it being a separate role. Let's at least put it in a separate tasks files and call it from post.yml though.

Reply:
Sure thing, these tasks will not be called if rgw is disabled as per this condition
- cifmw_ceph_daemons_layout.rgw_enabled | default(true) | bool